File: server/php/src/Application/Interfaces/SDKClients/HostedCheckoutClientInterface.php
<?php

namespace OnlinePayments\ExampleApp\Application\Interfaces\SDKClients;

use OnlinePayments\ExampleApp\Application\DTOs\HostedCheckout\RequestDto as CreateHostedCheckoutRequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\HostedCheckout\ResponseDto as CreateHostedCheckoutResponseDto;
use OnlinePayments\ExampleApp\Application\DTOs\GetPaymentByHostedCheckoutId\ResponseDto as GetPaymentByHostedCheckoutIdResponseDto;

interface HostedCheckoutClientInterface
{
    public function createHostedCheckout(CreateHostedCheckoutRequestDto $requestDto): CreateHostedCheckoutResponseDto;

    public function getPaymentByHostedCheckoutId(string $id): ?GetPaymentByHostedCheckoutIdResponseDto;
}

File: server/python/app/application/interfaces/sdk_clients/hosted_checkout_client_interface.py
from abc import ABC, abstractmethod
from app.application.dtos.hosted_checkout.request_dto import CreateHostedCheckoutRequestDto
from app.application.dtos.hosted_checkout.response_dto import CreateHostedCheckoutResponseDto
from app.application.dtos.get_payment_by_hosted_checkout_id.response_dto import GetPaymentByHostedCheckoutIdResponseDto

class HostedCheckoutClientInterface(ABC):

    @abstractmethod
    def create_hosted_checkout(self, request_dto: CreateHostedCheckoutRequestDto) -> CreateHostedCheckoutResponseDto:
        pass

    @abstractmethod
    def get_payment_by_hosted_checkout_id(self, id: str) -> GetPaymentByHostedCheckoutIdResponseDto:
        pass